src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -2217,3 +2230,81 @@ int32_t guest_path_finding(Guest* peep)
 #endif // defined(DEBUG_LEVEL_1) && DEBUG_LEVEL_1
     return peep_move_one_tile(direction, peep);
 }
+
+bool is_valid_path_z_and_direction(TileElement* tileElement, int32_t currentZ, int32_t currentDirection)
```

Comment:
Since you are moving these functions can you go through and follow the coding style. 
Function names should be TitleCase. variable names should be lower camelCase (which I'm pretty sure they all are).